### PR TITLE
WFLY-8418 licenses

### PR DESCRIPTION
**Pull request by psakar**
generate licenses.xml and html

issue: https://issues.jboss.org/browse/WFLY-8418

### CODE REVIEW

**Comment by kabir** [comment]
Retest this please

**Comment by kabir** [comment]
Retest this please

**Comment by psakar** [comment]
retest this please

File: dist/pom.xml [hunk 1]
```diff
@@ -184,6 +184,102 @@
...
+                                                <mapper type="flatten"/>
+                                                <globmapper from="*" to="&lt;item&gt;*"/>
+                                                <globmapper from="*" to="*&lt;/item&gt;"/>
+                                            </chainedmapper>
+                                        </pathconvert>
+                                        <echo file="${basedir}/target/${server.output.dir.prefix}-${jboss.as.release.version}/docs/licenses/licenses.xml">&lt;?xml version="1.0" encoding="UTF-8"?&gt;&#xA;&lt;list&gt;&#xA;</echo>
+                                        <echo file="${basedir}/target/${server.output.dir.prefix}-${jboss.as.release.version}/docs/licenses/licenses.xml" append="true">${license-files-list-items}&#xA;</echo>
+                                        <echo file="${basedir}/target/${server.output.dir.prefix}-${jboss.as.release.version}/docs/licenses/licenses.xml" append="true">&lt;/list&gt;</echo>
+                                    </tasks>
+                                </configuration>
```
Comment (psakar) [inline dist/pom.xml]: This part generates file which contains list of *licenses.xml files provisioned by feature packs (using ant task)

File: dist/pom.xml [hunk 2]
```diff
@@ -184,6 +184,102 @@
...
+                                            <dir>${basedir}/target/${server.output.dir.prefix}-${jboss.as.release.version}/docs/licenses</dir>
+                                            <includes>
+                                                <include>licenses.xml</include>
+                                            </includes>
+                                            <stylesheet>${basedir}/target/${server.output.dir.prefix}-${jboss.as.release.version}/docs/licenses/licenses-merged.xsl</stylesheet>
+                                            <outputDir>${basedir}/target/${server.output.dir.prefix}-${jboss.as.release.version}/docs/licenses</outputDir>
+                                            <fileMappers>
+                                                <fileMapper implementation="org.codehaus.plexus.components.io.filemappers.FileExtensionMapper">
+                                                    <targetExtension>.xml</targetExtension>
+                                                </fileMapper>
```
Comment (psakar) [inline dist/pom.xml]: This xslt transformation takes xml file with list of *licenses.xml files and merges their content into single licenses.xml file

File: dist/pom.xml [hunk 3]
```diff
@@ -184,6 +184,102 @@
...
+                                                    <targetExtension>.html</targetExtension>
+                                                </fileMapper>
+                                            </fileMappers>
+                                            <parameters>
+                                                <parameter>
+                                                    <name>version</name>
+                                                    <value>${project.version}</value>
+                                                </parameter>
+                                            </parameters>
+                                        </transformationSet>
```
Comment (psakar) [inline dist/pom.xml]: This xslt transformation takes single licenses.xml file which contains everything from all *licenses.xml files and converts it to licenses.html file

**Pull request merged**